config/server.py

```diff
@@ -0,0 +1,196 @@
+        if len(routes) != 1:
+            raise InvalidRouteID()
+        else:
+            return routes[0]
+
+    def _generate_new_token(self):
+        return str(uuid.uuid4())
+
+    # Swagger called functions
+
+    @_auth_request
+    def patch_route(self, token, new_info):
+        self._token2route(token).update(**new_info).execute()
+
+        return None, 204
+
+    @_auth_request
+    def delete_route(self, token):
+        try:
+            self._token2route(token).delete().execute()
+        except InvalidRouteID:
+            pass  # DELETE requests are supposed to be idempotent
+
+        return None, 204
+
+    def get_route(self, token):
+        return self._token2route(token).get_json()
+
+    @_auth_request
+    def get_all_routes(self):
+        routes = self.Route.select().where(self.Route.owner == self.auth.get_user())
+
+        return [route.get_json() for route in routes]
+
+    @_auth_request
+    def add_route(self, new_route):
+        try:
+            url_ob = urlparse(new_route["destination"])
+            if url_ob.scheme == '':
+                destination = "http://" + new_route["destination"]
+            else:
+                destination = new_route["destination"]
+        except SyntaxError:
+            raise InvalidURL()
+
+        route = self.Route(
+            owner=self.auth.get_user(),
+            destination=destination,
+            name=new_route["name"],
+            token=self._generate_new_token())
+
+        route.save()
+
+        return route.get_json(), 201
+
+    @_auth_request
+    def regenerate_token(self, token):
+        route = self._token2route(token)
+        new_token = self._generate_new_token()
+        route.update(token=new_token).execute()
+
+        return {
+            **route.get_json(),
+            "token": new_token
+        }
+
+    def close(self):
+        self.db.close()
+
+    def resolve_name(self, name):
+        return getattr(self, name)
+
+    def _set_error_handler(self, error_class, error_message, error_code):
+        def handler(error):
+            return flask.make_response(flask.jsonify({'error': error_message}), error_code)
+        self.app.add_error_handler(error_class, handler)
+
+    def __init__(self, debug, db, auth):
+        self.db = db
+        self.auth = auth
+        self.db.connect()
+        self.Route = get_route_model(self.db)
+        self.db.create_tables([self.Route], True)
+        self.app = connexion.FlaskApp(__name__, specification_dir=".", debug=debug)
+
+        self._set_error_handler(InvalidRouteID, "Invalid route ID", 404)
+        self._set_error_handler(NotAuthorised, "Not Authorised", 403)
+        self._set_error_handler(InvalidURL, "Invalid URL in destination", 400)
+        self._set_error_handler(InvalidCredentials, "Invalid credentials", 403)
+
+        self.app.add_api('swagger.yaml', resolver=connexion.Resolver(self.resolve_name), validate_responses=debug)
```

Comment:
There's likely little harm in always validating responses, in the same vein that you can (and perhaps should) run production code with assertions enabled. IMO detecting bugs and preventing invalid output is more important than tiny speed ups in execution.